# Практическая работа №2: Обработка выборочных данных. Нахождение точечных оценок параметров распределения

Выполнили студенты гр. 2381 Дудкин Михаил Валерьевич и Газукина Дарья Денисовна. Вариант №19.

## Цель работы

Получение практических навыков нахождения точечных статистических оценок параметров распределения.

## Основные теоретические положения

**Условные варианты** — преобразованные значения признака для упрощения вычислений с группированными данными:
$$
u_i = \frac{x_i - C}{h}
$$

**Точечная статистическая оценка** — числовая функция от выборки, используемая для оценивания неизвестного параметра генеральной совокупности:
$$
\hat{\theta} = \hat{\theta}(X_1, \dots, X_n)
$$

**Математическое ожидание** — среднее значение случайной величины, характеризующее центр её распределения:
$$
\mu = M[X] =
\begin{cases}
\displaystyle \sum_{i} x_i p_i, & \text{дискретная СВ} \\[10pt]
\displaystyle \int_{-\infty}^{+\infty} x f(x)\,dx, & \text{непрерывная СВ}
\end{cases}
$$

**Дисперсия** — мера рассеяния значений случайной величины относительно её математического ожидания:
$$
\sigma^2 = D[X] = M\left[(X - \mu)^2\right] = M[X^2] - (M[X])^2
$$

**Среднеквадратичное отклонение** — корень из дисперсии, мера рассеяния в единицах исходной величины:
$$
\sigma = \sqrt{D[X]}
$$

**Коэффициент асимметрии** — мера несимметричности распределения относительно математического ожидания:
$$
A_s = \frac{\mu_3}{\sigma^3}, \quad \mu_3 = M\left[(X - \mu)^3\right]
$$

**Коэффициент эксцесса** — мера островершинности распределения по сравнению с нормальным законом:
$$
E_k = \frac{\mu_4}{\sigma^4} - 3, \quad \mu_4 = M\left[(X - \mu)^4\right]
$$

**Мода** — значение случайной величины, соответствующее максимуму плотности распределения:
$$
M_o: \quad f'(M_o) = 0, \quad f''(M_o) < 0
$$

**Медиана** — значение, делящее распределение пополам: 50% наблюдений лежат ниже, 50% — выше:
$$
M_e: \quad F(M_e) = 0.5
$$

**Коэффициент вариации** — относительная мера рассеяния, позволяющая сравнивать изменчивость величин с разными единицами измерения:
$$
V = \frac{\sigma}{\mu} \cdot 100\%
$$

**Условный эмпирический момент $k$-го порядка** — момент, вычисленный по условным вариантам $u_i$ для упрощения расчётов:
$$
\bar{M}^*_k = \frac{1}{n} \sum_{i=1}^{k} n_i u_i^k
$$

**Центральный эмпирический момент $k$-го порядка** — момент относительно выборочного среднего, характеризующий форму распределения:
$$
\bar{m}_k = \frac{1}{n} \sum_{i=1}^{n} (x_i - \bar{x})^k
$$

**Связь центральных и условных моментов** — формулы перехода от условных моментов к центральным при $x_i = C + h u_i$:
$$
\begin{aligned}
\bar{m}_1 &= 0 \\
\bar{m}_2 &= h^2 \left( \bar{M}^*_2 - (\bar{M}^*_1)^2 \right) \\
\bar{m}_3 &= h^3 \left( \bar{M}^*_3 - 3\bar{M}^*_1 \bar{M}^*_2 + 2(\bar{M}^*_1)^3 \right) \\
\bar{m}_4 &= h^4 \left( \bar{M}^*_4 - 4\bar{M}^*_1 \bar{M}^*_3 + 6(\bar{M}^*_1)^2 \bar{M}^*_2 - 3(\bar{M}^*_1)^4 \right)
\end{aligned}
$$

## Постановка задачи

Для заданных выборочных данных вычислить с использованием метода моментов и условных вариант точечные статистические оценки математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации исследуемой случайной величины. Полученные результаты содержательно проинтерпретировать.

## Выполнение работы

Импорт необходимых библиотек

In [69]:
import math
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

Загрузка выборки

In [70]:
# Вариант №19
N = 103   # Объем выборки

In [71]:
path = "/content/sample_103.csv"
df = pd.read_csv(path)

# Задание 1

Для середин интервального ряда, полученного в практической работе №1, вычислить условные варианты. Заполнить табл. 1 (в последней строке $Σ$ необходимо заполнить суммы столбцов; ячейки отмеченные прочерком заполнять не надо). Провести контроль вычислений.

In [72]:
import pandas as pd
import numpy as np

data = df['nu'].values

# Количество интервалов (формула Стерджеса)
k = round(1 + 3.322 * np.log10(N))

# Шаг интервала
h = (data.max() - data.min()) / k
h = np.round(h, 2)

# Границы интервалов
bins = np.linspace(data.min(), data.max(), k + 1)
bins[-1] += 0.01  # Чтобы максимальное значение попало в последний интервал

# Середины интервалов
x_i = (bins[:-1] + bins[1:]) / 2

# Частоты
N_i, _ = np.histogram(data, bins=bins)
n_i = N_i / N
C_idx = np.argmax(n_i)
C = x_i[C_idx]

# Условные варианты:
u_i = np.round((x_i - C) / h).astype(int)

table1 = pd.DataFrame({
    'i': range(1, k + 1),
    'x_i': np.round(x_i, 2),
    'n_i': n_i,
    'u_i': u_i,
    '(n_i u_i)': n_i * u_i,
    '(n_i u_i^2)': n_i * u_i**2,
    '(n_i u_i^3)': n_i * u_i**3,
    '(n_i u_i^4)': n_i * u_i**4,
    '(n_i (u_i + 1)^4)': n_i * (u_i + 1)**4
})

sum_row = {
    'i': 'Σ',
    'x_i': '-',
    'n_i': table1['n_i'].sum(),
    'u_i': '-',
    '(n_i u_i)': table1['(n_i u_i)'].sum(),
    '(n_i u_i^2)': table1['(n_i u_i^2)'].sum(),
    '(n_i u_i^3)': table1['(n_i u_i^3)'].sum(),
    '(n_i u_i^4)': table1['(n_i u_i^4)'].sum(),
    '(n_i (u_i + 1)^4)': table1['(n_i (u_i + 1)^4)'].sum()
}
table1 = pd.concat([table1, pd.DataFrame([sum_row])], ignore_index=True)

print("ТАБЛИЦА №1")
table1


ТАБЛИЦА №1


,i,x_i,n_i,u_i,(n_i u_i),(n_i u_i^2),(n_i u_i^3),(n_i u_i^4),(n_i (u_i + 1)^4)
0,1,345.94,0.038835,-4,-0.155340,0.621359,-2.485437,9.941748,3.145631
1,2,377.81,0.097087,-3,-0.291262,0.873786,-2.621359,7.864078,1.553398
2,3,409.69,0.155340,-2,-0.310680,0.621359,-1.242718,2.485437,0.155340
3,4,441.56,0.203883,-1,-0.203883,0.203883,-0.203883,0.203883,0.000000
4,5,473.44,0.262136,0,0.000000,0.000000,0.000000,0.000000,0.262136
5,6,505.31,0.126214,1,0.126214,0.126214,0.126214,0.126214,2.019417
6,7,537.19,0.067961,2,0.135922,0.271845,0.543689,1.087379,5.504854
7,8,569.07,0.048544,3,0.145631,0.436893,1.310680,3.932039,12.427184
8,Σ,-,1.000000,-,-0.553398,3.155340,-4.572816,25.640777,25.067961


In [73]:
control_sum = table1['(n_i (u_i + 1)^4)'].iloc[-1]
moment_sum = table1['(n_i u_i^4)'].iloc[-1] + 4 * table1['(n_i u_i^3)'].iloc[-1] + 6 * table1['(n_i u_i^2)'].iloc[-1] + 4 * table1['(n_i u_i)'].iloc[-1] + 1
print(f'{control_sum} = {moment_sum}')

25.067961165048544 = 25.06796116504854


Вычислены условные варианты по интервальному ряду выборки. Найдены данные для заполнения таблицы 1. Сумма столбца $n_i$ равна 1, так как в этом столбце записаны относительный частоты для условных вариант. Контроль вычислений произведен по тождеству $\sum n_j(u_j + 1)^4 = \sum n_j u_j^4 + 4\sum n_j u_j^3 + 6\sum n_j u_j^2 + 4\sum n_j u_j + 1$

# Задание 2

Вычислить условные эмпирические моменты $\bar{M}^*_k$ через условные варианты. С помощью условных эмпирических моментов вычислить центральные эмпирические моменты $\bar{m}_k$. Полученные результаты занести в табл. 2.

In [74]:
M_star_1 = table1['(n_i u_i)'].iloc[-1]
M_star_2 = table1['(n_i u_i^2)'].iloc[-1]
M_star_3 = table1['(n_i u_i^3)'].iloc[-1]
M_star_4 = table1['(n_i u_i^4)'].iloc[-1]

m_1 = 0

m_2 = h**2 * (M_star_2 - M_star_1**2)

m_3 = h**3 * (M_star_3 - 3 * M_star_1 * M_star_2 + 2 * M_star_1**3)

m_4 = h**4 * (M_star_4 - 4 * M_star_1 * M_star_3 + 6 * M_star_1**2 * M_star_2 - 3 * M_star_1**4)

table2 = pd.DataFrame({
    'k': [1, 2, 3, 4],
    'M*_k': [M_star_1, M_star_2, M_star_3, M_star_4],
    'm_k': [m_1, m_2, m_3, m_4]
})

print("ТАБЛИЦА №2")
table2

ТАБЛИЦА №2


,k,M*_k,m_k
0,1,-0.553398,0.000000e+00
1,2,3.155340,2.895629e+03
2,3,-4.572816,1.058550e+04
3,4,25.640777,2.172779e+07


Условные эмпирические моменты получены, как суммы соответствующих столбцов таблицы 1. Центральные эмпирические моменты найдены по формулам:

$\bar{m}_1 = 0$

$\bar{m}_2 = \left(\bar{M}_2^* - \left(\bar{M}_1^*\right)^2\right)h^2$

$\bar{m}_3 = \left(\bar{M}_3^* - 3\bar{M}_2^*\bar{M}_1^* + 2\left(\bar{M}_1^*\right)^3\right)h^3$

$\bar{m}_4 = \left(\bar{M}_4^* - 4\bar{M}_3^*\bar{M}_1^* + 6\bar{M}_2^*\left(\bar{M}_1^*\right)^2 - 3\left(\bar{M}_1^*\right)^4\right)h^4$

# Задание 3

Вычислить выборочные среднее $\bar{x}_s$ и дисперсию $D_s$ с помощью стандартной формулы и с помощью условных вариант. Убедиться, что результаты совпадают. Вычислить выборочное СКО $\sigma_s$.

In [75]:
x_mean = np.sum(n_i * x_i)

D_s = np.sum(n_i * (x_i - x_mean) ** 2)

sigma_s = np.sqrt(D_s)

u_mean = table1['(n_i u_i)'].iloc[-1]

# Выборочное среднее через условные варианты: x̄ = C + h * ū
x_mean_conditional = C + h * u_mean

D_s_conditional = m_2

print("ВЫБОРОЧНЫЕ ХАРАКТЕРИСТИКИ")

print(f"Выборочное среднее x_bar:")
print(f"Стандартная формула: {x_mean:.4f}")
print(f"Через условные варианты: {x_mean_conditional:.4f}")
print(f"Разница: {abs(x_mean - x_mean_conditional) / x_mean:.2e} \n")

print(f"Выборочная дисперсия Dв:")
print(f"Стандартная формула: {D_s:.4f}")
print(f"Через условные варианты: {D_s_conditional:.4f}")
print(f"Разница: {abs(D_s - D_s_conditional) / D_s:.2e} \n")

print(f"Выборочное СКО sigma_s: {sigma_s:.4f}")


ВЫБОРОЧНЫЕ ХАРАКТЕРИСТИКИ
Выборочное среднее x_bar:
Стандартная формула: 455.7982
Через условные варианты: 455.7952
Разница: 6.60e-06 

Выборочная дисперсия Dв:
Стандартная формула: 2894.7753
Через условные варианты: 2895.6286
Разница: 2.95e-04 

Выборочное СКО sigma_s: 53.8031


Вычислены выборочное среднее и выборочная дисперсия по исходным вариантам интервального ряда и с помощью условных вариант. Значения совпадают с высокой точностью, что говорит o работоспособности метода моментов и условных вариант.

# Задание 4

Вычислить исправленную выборочную дисперсию $s^2$ и исправленное СКО $s$. Сравнить данные оценки с смещёнными оценками дисперсии и СКО. Сделать выводы.

In [76]:
s2 = D_s * N / (N - 1)
s = np.sqrt(s2)
sigma_s = np.sqrt(D_s)

print("ИСПРАВЛЕННЫЕ ОЦЕНКИ")
print(f"s^2 = {s2:.4f}  (смещённая Dв = {D_s:.4f})")
print(f"s  = {s:.4f}   (смещённая sigma_в = {sigma_s:.4f})")

ИСПРАВЛЕННЫЕ ОЦЕНКИ
s^2 = 2923.1555  (смещённая Dв = 2894.7753)
s  = 54.0662   (смещённая sigma_в = 53.8031)


Исправленная дисперсия $s^2 = \frac{N}{N-1} \cdot D_в$ отличается от смещённой $D_в$ на поправочный множитель Бесселя $\frac{N}{N-1}$.

При $N = 103$
поправка: $\frac{103}{102} \approx 1.0098$

При больших $N$ смещение незначительно, но для корректных статистических выводов следует использовать несмещённые оценки $s^2$ и $s$.

# Задание 5

Найти статистическую оценку коэффициентов асимметрии $\bar{A}_s$ и эксцесса $\bar{E}$. Сделать выводы.

In [77]:
A_s = m_3 / s**3
E = m_4 / s**4 - 3

print("КОЭФФИЦИЕНТЫ ФОРМЫ РАСПРЕДЕЛЕНИЯ")
print(f"Асимметрия A_s = {A_s:.4f}")
print(f"Эксцесс E = {E:.4f}")

КОЭФФИЦИЕНТЫ ФОРМЫ РАСПРЕДЕЛЕНИЯ
Асимметрия A_s = 0.0670
Эксцесс E = -0.4572


Выборочные коэффициенты формы:
- Асимметрия $\bar{A}_s = \frac{\bar{m}_3}{s^3} = 0.067$ — распределение симметрично
- Эксцесс $\bar{E} = \frac{\bar{m}_4}{s^4} - 3 = -0.4572$ — незначительное плосковершинное отклонение

Так как $|\bar{A}_s| < 0.5$ и $|\bar{E}| < 0.5$, эмпирическое распределение близко к нормальному.


#  Задание 6

Для интервального ряда вычислить моду $M^*_o$, медиану $M^*_e$ и коэффициент вариации $V^*$ заданного распределения. Сделать выводы.

In [78]:
modal_idx = np.argmax(n_i)  # индекс модального интервала
n_Mo = n_i[modal_idx]
n_Mo_prev = n_i[modal_idx - 1] if modal_idx > 0 else 0
n_Mo_next = n_i[modal_idx + 1] if modal_idx < len(n_i) - 1 else 0
x_Mo_lower = bins[modal_idx]  # нижняя граница модального интервала

M_o = x_Mo_lower + h * (n_Mo - n_Mo_prev) / ((n_Mo - n_Mo_prev) + (n_Mo - n_Mo_next))


cum_w = np.cumsum(n_i)
median_idx = np.searchsorted(cum_w, 0.5)  # индекс медианного интервала
S_prev = cum_w[median_idx - 1] if median_idx > 0 else 0
x_Me_lower = bins[median_idx]
n_Me = n_i[median_idx]

M_e = x_Me_lower + h * (0.5 - S_prev) / n_Me


V = s / x_mean_conditional * 100  # в процентах

print("РОБАСТНЫЕ ХАРАКТЕРИСТИКИ ИНТЕРВАЛЬНОГО РЯДА")
print(f"Мода M*_o = {M_o:.4f}")
print(f"Медиана M*_e = {M_e:.4f}")
print(f"Среднее x_ = {x_mean_conditional:.4f}")
print(f"Коэфф. вариации V* = {V:.2f}%")

РОБАСТНЫЕ ХАРАКТЕРИСТИКИ ИНТЕРВАЛЬНОГО РЯДА
Мода M*_o = 467.0640
Медиана M*_e = 458.0904
Среднее x_ = 455.7952
Коэфф. вариации V* = 11.86%


- Мода $M_o^* = 467.06$ и медиана $M_e^* = 458.09$ близки между собой и к выборочному среднему, что подтверждает симметричность распределения.
- Коэффициент вариации $V^* = 11.86\%$ указывает на умеренную изменчивость, совокупность считается однородной.
- Робастные характеристики согласуются с ранее полученными оценками асимметрии ($\bar{A}_s = 0.067$) и эксцесса ($\bar{E} = -0.4572$).

# Выводы

В ходе выполнения практической работы были получены практические навыки нахождения точечных статистических оценок параметров распределения для выборки объёмом.

Метод условных вариант успешно применён для упрощения вычислений. Контроль по тождеству $\sum n_i(u_i+1)^4 = \sum n_i u_i^4 + 4\sum n_i u_i^3 + 6\sum n_i u_i^2 + 4\sum n_i u_i + \sum n_i$ подтвердил корректность расчётов. Сравнение методов вычисления показало, что выборочное среднее и дисперсия, найденные через стандартные формулы и через условные варианты, совпадают с высокой точностью (относительная погрешность $< 10^{-4}$), что подтверждает работоспособность метода моментов.

| Параметр | Значение | Интерпретация |
|----------|----------|---------------|
| $\bar{x}$ | 455.80 | Центр распределения |
| $s^2$ | 2923.16 | Исправленная дисперсия |
| $s$ | 54.07 | Исправленное СКО |
| $\bar{A}_s$ | 0.067 | Распределение практически симметрично |
| $\bar{E}_k$ | -0.457 | Слабое плосковершинное отклонение от нормального закона |
| $M_o^*$ | 467.06 | Наиболее вероятное значение |
| $M_e^*$ | 458.09 | 50% данных лежат ниже этого значения |
| $V^*$ | 11.86% | Умеренная изменчивость, совокупность однородна |

Сравнение смещённых и несмещённых оценок показало, что исправленная дисперсия $s^2$ отличается от смещённой $D_в$ на множитель Бесселя $\frac{n}{n-1} \approx 1.0098$. При $n = 103$ разница незначительна (~1%), однако для корректных статистических выводов рекомендуется использовать несмещённые оценки.

Анализ формы распределения показал близость моды, медианы и среднего ($M_o^* \approx M_e^* \approx \bar{x}$), а также малые значения коэффициентов асимметрии ($|\bar{A}_s| < 0.5$) и эксцесса ($|\bar{E}_k| < 0.5$). Это свидетельствует о том, что эмпирическое распределение близко к нормальному закону.

Освоены методы нахождения точечных оценок параметров распределения с использованием метода моментов и условных вариант. Полученные результаты согласованы между собой и позволяют сделать вывод о симметричности и близости к нормальному распределению исследуемой случайной величины. Метод условных вариант доказал свою эффективность для упрощения расчётов с группированными данными без потери точности.
